src/CMakeLists.txt

```diff
@@ -18,10 +18,12 @@ else()
 endif()
 
 add_library(benchmark ${SOURCE_FILES} ${RE_FILES})
+find_package(Threads REQUIRED)
 
-# Link threading if building a shared library.
+# Link threading if building a shared library, otherwise let client do it.
 if (BUILD_SHARED_LIBS)
-  find_package(Threads REQUIRED)
+  target_link_libraries(benchmark PRIVATE ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
Actually, the cmake documentation seems to be saying that PUBLIC/PRIVATE/INTERFACE is only relevant for shared libraries (and dependencies are always propagated when linking statically), so I think this can be simplified a lot. What do you think of this?